### PR TITLE
feat(await-async-utils): reflect waitFor changes

### LINKED ISSUE #85: Update await-async-utils with changes from dom-testing-library v7
After release of [dom-testing-library v7](https://github.com/testing-library/dom-testing-library/releases/tag/v7.0.0) there are couple of things to be updated for `await-async-utils` rule:

- include `waitFor` as valid async util
- update the doc to mention `wait`, `waitForElement` and `waitForDomChange` are now deprecated
- make sure when `waitForElementToBeRemoved` receives an element rather than callback doesn't cause false positives

### PR DESCRIPTION
**Pull request by Belco90**
This closes #85

### CODE REVIEW

File: docs/rules/await-async-utils.md [hunk 1]
```diff
@@ -6,10 +6,11 @@ Ensure that promises returned by async utils are handled properly.
 
 Testing library provides several utilities for dealing with asynchronous code. These are useful to wait for an element until certain criteria or situation happens. The available async utils are:
 
-- `wait`
-- `waitForElement`
-- `waitForDomChange`
+- `waitFor` _(introduced in dom-testing-library v7)_
```
Comment (emmenko) [inline docs/rules/await-async-utils.md]: 💯

File: docs/rules/await-async-utils.md [hunk 2]
```diff
@@ -54,16 +52,13 @@ test('something correctly', async () => {
   );
 
   // `then` chained method is correct
-  waitForDomChange(() => {
-    return { container };
-  })
+  waitFor(() => {}, { timeout: 100 })
```
Comment (thomlom) [inline docs/rules/await-async-utils.md]: Why adding a `timeout` to the example? 🤔

Comment (Belco90) [reply docs/rules/await-async-utils.md]: It was just a stupid config object. I updated it on tests, I can do the same here.

Comment (Belco90) [reply docs/rules/await-async-utils.md]: No, sorry. It's actually fine like this. This is a real example from new [`waitFor` optional second argument](https://testing-library.com/docs/dom-testing-library/api-async#waitfor).

**Pull request merged**